api/rpc/account/account.go

```diff
@@ -0,0 +1,311 @@
+type Server struct{}
+
+// SignUp implements account.SignUp
+func (s *Server) SignUp(ctx context.Context, in *SignUpRequest) (out *SessionReply, err error) {
+	out = &SessionReply{}
+	err = in.Validate()
+	if err != nil {
+		return nil, err
+	}
+	_, err = passlib.Hash(in.Password)
+	if err != nil {
+		return nil, grpc.Errorf(codes.Internal, "hashing error")
+	}
+	out.SessionKey = in.Name
+	return
+}
+
+// Verify implements account.Verify
+func (s *Server) Verify(ctx context.Context, in *VerificationRequest) (out *pb.Empty, err error) {
+	out = &pb.Empty{}
+	err = in.Validate()
+	if err != nil {
+		return nil, err
+	}
+	fmt.Println(in.Code)
+	return
+}
+
+// CreateOrganization implements account.CreateOranization
+func (s *Server) CreateOrganization(ctx context.Context, in *OrganizationRequest) (out *pb.Empty, err error) {
+	out = &pb.Empty{}
+	err = in.Validate()
+	if err != nil {
+		return nil, err
+	}
+	return
+}
+
+// Login implements account.Login
+func (s *Server) Login(ctx context.Context, in *LogInRequest) (out *SessionReply, err error) {
+	out = &SessionReply{}
+	err = in.Validate()
+	if err != nil {
+		return nil, err
+	}
+	_, err = passlib.Verify(in.Password, hash)
+	if err != nil {
+		return nil, grpc.Errorf(codes.Unauthenticated, err.Error())
+	}
+	out.SessionKey = in.Name
+	return
+
+}
+
+// Switch implements account.Switch
+func (s *Server) Switch(ctx context.Context, in *TeamRequest) (out *pb.Empty, err error) {
+	out = &pb.Empty{}
+	if in.Organization == "" {
+		return nil, grpc.Errorf(codes.InvalidArgument, "organization name is mandatory")
+	}
+	return
+}
+
+// ListAccounts implements account.ListAccounts
+func (s *Server) ListAccounts(ctx context.Context, in *AccountsRequest) (out *ListReply, err error) {
+	out = &ListReply{}
+	if in.Type != "individual" && in.Type != "organization" {
+		return nil, grpc.Errorf(codes.InvalidArgument, "account type is mandatory")
+	}
+	return
+}
+
+// GetAccountDetails implements account.GetAccountDetails
+func (s *Server) GetAccountDetails(ctx context.Context, in *AccountRequest) (out *AccountReply, err error) {
+	out = &AccountReply{}
+	if in.Name == "" {
+		return nil, grpc.Errorf(codes.InvalidArgument, "name is mandatory")
+	}
+	return
+}
+
+// EditAccount implements account.EditAccount
+func (s *Server) EditAccount(ctx context.Context, in *EditRequest) (out *AccountReply, err error) {
+	out = &AccountReply{}
+	err = in.Validate()
+	if err != nil {
+		return
+	}
+	if in.NewPassword != "" {
+		_, err := passlib.Verify(in.Password, hash)
+		if err != nil {
+			return nil, grpc.Errorf(codes.Unauthenticated, err.Error())
+		}
+		_, err = passlib.Hash(in.NewPassword)
+		if err != nil {
+			return nil, grpc.Errorf(codes.Internal, "hashing error")
+		}
+	}
+	return
+}
+
+// DeleteAccount implements account.DeleteAccount
+func (s *Server) DeleteAccount(ctx context.Context, in *AccountRequest) (out *pb.Empty, err error) {
+	out = &pb.Empty{}
+	if in.Name == "" {
```

Comment:
I like that you moved all the validators out to their own file. Regarding using validators, it seems like we should be consistent in applying the pattern. Thus, we probably want a Validate method on AccountRequest as well for standardization, even though the validation is so simple.  The same principle would apply to all the following request messages, such as TeamRequest, where the same validation logic keeps getting applied (we may want to create a validation context or enum parameter for CRUD-type operations that validate different subsets of the message fields if we don't want to create distinct message types).